### PR TITLE
added gitignore entries for clion and qtcreator ides, plus some libra…

### PR DESCRIPTION
**Pull request by ALEXJAZZ008008**
…ry files which were missing. can comment idea file as prefered.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -184,6 +184,81 @@ UpgradeLog*.htm
...
+object_script.*.Debug
+*_plugin_import.cpp
+/.qmake.cache
+/.qmake.stash
+*.pro.user
+*.pro.user.*
+*.qbs.user
+*.qbs.user.*
+*.moc
+moc_*.cpp
```
Comment (KrisThielemans) [inline .gitignore]: somewhat reluctant to add these. who knows if we ever call a file that way?

File: .gitignore [hunk 2]
```diff
@@ -184,6 +184,81 @@ UpgradeLog*.htm
...
+moc_*.h
+qrc_*.cpp
+ui_*.h
+*.qmlc
+*.jsc
+Makefile*
+*build-*
+
+# Qt unit tests
+target_wrapper.*
```
Comment (KrisThielemans) [inline .gitignore]: same here

File: .gitignore [hunk 3]
```diff
@@ -184,6 +184,81 @@ UpgradeLog*.htm
...
 FakesAssemblies/
 
+# Covers JetBrains IDEs: CLion
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# Ignore entire idea
+# Comment as required
+*.idea
+
+# User-specific stuff
```
Comment (KrisThielemans) [inline .gitignore]: what about just saying: `.idea`?

Comment (bathomas) [reply .gitignore]: For CLion I tend to use:

```
.idea/*
cmake-build-debug/*
cmake-build-release/*
```

**Comment by KrisThielemans** [comment]
@ALEXJAZZ008008 could you finish this one please?

**Comment by ALEXJAZZ008008** [comment]
Sorry, I wasn't certain if we were still interested in merging the edits I had made to the .gitignore.
I've removed the first two blocks you've mentioned and edited the third block based on Ben's comment.
Regarding Ben's comment, why is it more advantageous to ignore the contents of the .idea directory rather than just ignore any directory with the prefix .idea?

**Comment by bathomas** [comment]
I had a problem a while back that the IDE was still showing local changes to files within `.idea` even when `.idea` had been added to `.gitignore`. Ignoring the contents appeared to fix it.

Thinking about it, possibly it would be better to have `cmake-build-*/` ?

**Comment by ALEXJAZZ008008** [comment]
I agree that cmake-build-*/* would be better, I've edited the lines.

File: .gitignore [hunk 4]
```diff
@@ -184,6 +184,73 @@ UpgradeLog*.htm
...
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+.idea/**/dbnavigator.xml
+
+# CMake
+cmake-build-*/
```
Comment (KrisThielemans) [inline .gitignore]: repetition

File: .gitignore [hunk 5]
```diff
@@ -184,6 +184,73 @@ UpgradeLog*.htm
...
 
+# Covers JetBrains IDEs: CLion
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# CLion
+.idea/*
+cmake-build-*/*
+
+# User-specific stuff
+.idea/**/workspace.xml
```
Comment (KrisThielemans) [inline .gitignore]: seems we don't need all of these anymore as far as I know

File: .gitignore [hunk 6]
```diff
@@ -184,6 +184,73 @@ UpgradeLog*.htm
...
+*.iws
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
+
+# Editor-based Rest Client
+.idea/httpRequests
```
Comment (KrisThielemans) [inline .gitignore]: not necessary anymore

File: .gitignore (hunk 3 above)
Comment (KrisThielemans) [reply .gitignore]: git doc says to use `.idea/` etc. Please check

**Pull request closed without merging**